.devops/code-review-pipelines.yml

```diff
@@ -0,0 +1,86 @@
+# Azure DevOps pipeline to build, check source codes and run tests.
+#
+# To make Danger JS run on a pull request you need to add the following pipeline 
+# variable and set it with a GitHub access token (scope public_repo); otherwise 
+# set its value to 'skip' without marking it secret: 
+# - DANGER_GITHUB_API_TOKEN
+# 
+
+variables:
+  NODE_VERSION: '8.9.4'
+  YARN_CACHE_FOLDER: $(Pipeline.Workspace)/.yarn
+
+# Automatically triggered on PR
+# https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#pr-trigger
+trigger: none
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+stages:
+  - stage: Build
+    dependsOn: []
+    jobs:
+      - job: make_build
+        steps:
+          - template: azure-templates/setup-project.yml
+          - script: |
+              yarn build
+            displayName: 'Build'
+  
+  - stage: Static_analysis
+    dependsOn: []
+    jobs:          
+      - job: lint
+        steps:
+          - template: azure-templates/setup-project.yml
+          - script: |
+              yarn lint
+            displayName: 'Lint'
+
+      - job: danger
+        condition: 
+          and(
+            succeeded(), 
+            and(
+              eq(variables['Build.Reason'], 'PullRequest'),
+              ne(variables['DANGER_GITHUB_API_TOKEN'], 'skip')
+            )
+          )
+        steps:
+          - template: azure-templates/setup-project.yml
+          - bash: |
+              yarn danger ci
+            env:
+              DANGER_GITHUB_API_TOKEN: '$(DANGER_GITHUB_API_TOKEN)'
+            displayName: 'Danger CI'
+
+  - stage: Test
+    pool:
+      vmImage: 'ubuntu-latest'
+    dependsOn: []
+    jobs:
+      - job: unit_tests
+        steps:
+          - template: azure-templates/setup-project.yml
+        
+          - script: |
+              yarn generate
+            displayName: 'Generate definitions'        
+        
+          - script: |
+              yarn test:unit
+            displayName: 'Unit tests'
+
+          - script: |
+              docker run -d -p 6379:6379 --name redis redis
+            displayName: 'Start redis'
+
+          - script: |
+              yarn test:integration
+            displayName: 'Unit tests integration'
+
+          - bash: |
+              bash <(curl -s https://codecov.io/bash)
+            displayName: 'Code coverage'
```

Comment:
This is not going to work unless we generate coverage reports from tests. This is as easy as add `--coverage` to test script. However, it must be noted that we use two different test scripts (unit and integration). I don't know how to make two tests contribute to the same coverage, if it's even possibile.

I see we might somehow merge two different reports (see [here](https://medium.com/@kushmisra7/one-report-for-all-test-cases-easily-merging-multiple-tests-reports-b0f5e5211a2a)), however I think that's really out of scope for this PR.

Meanwhile, what we can do is to generate coverage for unit tests only. Furthermore, we can think about move integration test in a dedicated stage, so it can be executed in parallel (not sure it's a real gain in performance, must give it a try)

Reply:
I splitted Unit and Integration tests, and I changed unit tests script to _test:coverage_